pkg/crc/oc/oc.go

```diff
@@ -0,0 +1,48 @@
+package oc
+
+import (
+	"fmt"
+	"path/filepath"
+	"strings"
+
+	"github.com/code-ready/crc/pkg/crc/constants"
+	crcos "github.com/code-ready/crc/pkg/os"
+)
+
+type OC struct {
+	OcBinaryPath   string
+	KubeconfigPath string
+}
+
+func GetOC(machineName string) OC {
+	oc := OC{
+		OcBinaryPath:   filepath.Join(constants.OcCacheDir, constants.OcBinaryName),
+		KubeconfigPath: filepath.Join(constants.MachineInstanceDir, machineName, "kubeconfig"),
+	}
+	return oc
+}
+
+func (oc OC) RunOcCommand(args ...string) (string, string, error) {
+	args = append(args, "--kubeconfig", oc.KubeconfigPath)
+	return crcos.RunWithDefaultLocale(oc.OcBinaryPath, args...)
+}
+
+func ApproveCSR(oc OC) error {
```

Comment:
`ApproveCSR` for an `OC Config`? ?

Reply:
When we talk about `oc` it feels like the command, but here we talk about an `environment` or `configuration`. I would prefer to see that in the naming to avoid this confusion

Reply:
Ah it should be `ApproveNodeCSR` since we are approving the node csr value.

Reply:
also... but it is more how `OC` is passed in that sounds merely like a `oc` command wrapper, while it is a configed instance of the `oc` command, right? What aboud `OCConfig`.